src/charm.py

```diff
@@ -442,6 +449,9 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Event Handler for config changed event."""
         self._handle_status_update_config(event)
 
+    def _cleanup(self, event: RemoveEvent) -> None:
+        self.network_policy_handler.delete_ingress_policies()
```

Comment:
It might be a juju bug, but when I try removing kratos it gets into terminated state with unit lost. When I use `--force --no-wait`, the charm is removed but in both cases the network policy is not deleted.

Reply:
So I tried testing it a little more and when I test it manually, the policy is removed. I wrote an integration test to validate this and in the test the policy is not removed. This is very weird.

Reply:
As far as I can tell the stop event is never fired, this is most likely a juju bug and not something we can control. Let's leave this as is for now since it is the best we can do and juju will (hopefully) fix it in a later version.